### ISSUE TITLE
Feature request: removing simulators

### ISSUE THREAD

**Issue by Finesse** [vashpan/xcode-dev-cleaner#21]
Hi! Thanks for the wonderful tool. But there is a room for a big improvement.

Xcode simulators (don't confuse with virtual devices) take several gigabytes each. You can download them in Xcode settings:

![image](https://user-images.githubusercontent.com/9006227/110560385-a09fcb80-8191-11eb-9113-b1cdabb2fda6.png)

They are located at `/Library/Developer/CoreSimulator/Profiles/Runtimes`. The Xcode UI doesn't offer an option to delete them, the only way to do it is going the directory and removing them manually. It will be great if DevCleaner is able to remove them too.

**Comment by jacobweber** [comment]
+1

**Comment by vashpan** [comment]
I'm working on cleaning some simulator stuff for 2.3.0, but unfortunately not this particular stuff (but I'm open for it in the future)

First: DevCleaner works in sandbox, therefore `/Library/Developer` folder is different than user one that it had access to: `~/Library/Developer`- I would need to ask separate permission for that which would complicate app a bit.

Second: It requires root access to delete files from that folder, which is another layer of complication.

**Comment by Finesse** [comment]
@vashpan What if the application says something like?

> There can be more stuff at `/Library/Developer/CoreSimulator/Profiles/Runtimes` (a clickable button that opens the folder in Finder). Dev Cleaner can't access it, so you need to clean it manually

At least users will know that there is more stuff that they can delete and where to find this stuff.

**Comment by vashpan** [comment]
Yeah, we could think about something like that 👍 I'm slowly working on deleting simulators straight from DevCleaner, this information could be presented for example after cleaning simulators.

**Comment by martin-braun** [comment]
My `$HOME/Library/Developer/CoreSimulator/Devices` has almost 40GB. I wish this tool was the one to rule them all. I'd love to see progress in that regard.

**Comment by malhal** [comment]
> My `$HOME/Library/Developer/CoreSimulator/Devices` has almost 40GB. I wish this tool was the one to rule them all. I'd love to see progress in that regard.

I'd like this feature too. I had to use DaisyDisk to find and remove this to free up space for the Xcode beta.

**Comment by gahms** [comment]
Could a way to do this be to use `simctl` instead of the raw files? 

I have this snippet that I run periodically:
```
xcrun simctl list -j devices | jq '.["devices"][][] | select(.isAvailable ==false) | .udid' | xargs -n1 xcrun simctl delete
```
I am not sure this is the same as fills up in `/Library/Developer/CoreSimulator/Profiles/Runtimes` but it sure frees up space in `~/Library/Developer/CoreSimulator/Devices`

**Comment by vashpan** [comment]
> Could a way to do this be to use `simctl` instead of the raw files?

DevCleaner is running in sandbox, hence running external shell scripts is extremely limited unfortunately. Because yes, this will be the best and most clean way of doing this.

I'm considering a version of DevCleaner that will run outside of the AppStore & sandbox environment, so in the future this might be the way.

**Comment by markst** [comment]
This will be good, since I appear to have two directories:

<img width="1001" alt="Image" src="https://github.com/user-attachments/assets/f47e640a-e8ef-468b-9999-642bdbc6381f" />

<img width="590" alt="Image" src="https://github.com/user-attachments/assets/22fb9394-edf7-407c-8dc6-3e6ac8db1173" />

**Comment by markst** [comment]
https://developer.apple.com/forums/thread/766513

**Comment by vashpan** [comment]
Simulators are also used for other things, like Swift Previews etc, so it's not that straightforward I would say. Although those I think are in a different folder.

It's also possible that some older, unused device was left behind for some reason - so it could be a bug.

**Comment by bysiber** [comment]
+1 for this feature. Xcode simulators at `/Library/Developer/CoreSimulator/Profiles/Runtimes` can easily take 5-15 GB per runtime, and old ones are rarely cleaned up.

**Quick commands to manage simulators:**

```bash
# See all installed simulator runtimes and sizes
xcrun simctl runtime list -v

# Delete unavailable/old simulators
xcrun simctl delete unavailable

# Check total simulator runtime storage
du -sh /Library/Developer/CoreSimulator/Profiles/Runtimes/*

# Delete specific old runtimes (e.g., iOS 16)
# First find the identifier:
xcrun simctl runtime list
# Then delete it:
sudo xcrun simctl runtime delete <identifier>
```

Also worth checking the device support files, which accumulate with each iOS version you test on:

```bash
# Device support files (keep only your current iOS)
du -sh ~/Library/Developer/Xcode/iOS\ DeviceSupport/*
du -sh ~/Library/Developer/Xcode/watchOS\ DeviceSupport/*

# Remove old versions
rm -rf ~/Library/Developer/Xcode/iOS\ DeviceSupport/16.*
```

For keeping tabs on all these Xcode-related caches (simulators, DerivedData, device support, archives, and more), [ClearDisk](https://github.com/bysiber/cleardisk) monitors these paths from the menu bar — it's a free macOS utility that tracks 44+ developer cache locations.

**Comment by dkechag** [comment]
I can delete old simulators from Xcode fine by pressing the (i) button, but in my case I have 2 mounted old watchOS simulators that had been somehow left over. Mount can see them:
```
/dev/disk9s1 on /Library/Developer/CoreSimulator/Volumes/watchOS_22T572 (apfs, sealed, local, nodev, nosuid, read-only, journaled, noatime, nobrowse)
/dev/disk13s1 on /Library/Developer/CoreSimulator/Volumes/watchOS_23S303 (apfs, sealed, local, nodev, nosuid, read-only, journaled, noatime, nobrowse)
```

 diskutil can see them:
```
+-- Container disk6 D99AA294-28D2-4241-9BF7-B0723D65EDAE
|   ====================================================
|   APFS Container Reference:     disk6
|   Size (Capacity Ceiling):      4857004032 B (4.9 GB)
|   Capacity In Use By Volumes:   4696719360 B (4.7 GB) (96.7% used)
|   Capacity Not Allocated:       160284672 B (160.3 MB) (3.3% free)
|   |
|   +-< Physical Store disk4s1 8864025C-0384-4081-94EB-7781C12F6F06
|   |   -----------------------------------------------------------
|   |   APFS Physical Store Disk:   disk4s1
|   |   Size:                       4857004032 B (4.9 GB)
|   |
|   +-> Volume disk6s1 64BD56F7-DD94-4F55-93B6-F8A5C9EAE3B5
|       ---------------------------------------------------
|       APFS Volume Disk (Role):   disk6s1 (No specific role)
|       Name:                      WatchOS 11.5 Simulator Bundle (Case-insensitive)
|       Mount Point:               /Library/Developer/CoreSimulator/Cryptex/Images/bundle/SimRuntimeBundle-B1D0A814-3BD1-4826-84EE-EED352F893EB
|       Capacity Consumed:         4677017600 B (4.7 GB)
|       Sealed:                    No
|       FileVault:                 No
|
+-- Container disk9 15F31F3C-3853-44E5-AB8E-FB86122E542C
|   ====================================================
|   APFS Container Reference:     disk9
|   Size (Capacity Ceiling):      11404312576 B (11.4 GB)
|   Capacity In Use By Volumes:   11085893632 B (11.1 GB) (97.2% used)
|   Capacity Not Allocated:       318418944 B (318.4 MB) (2.8% free)
|   |
|   +-< Physical Store disk8s1 A3721E36-B2C6-4D5F-A5C3-43E0A26FB6E2
|   |   -----------------------------------------------------------
|   |   APFS Physical Store Disk:   disk8s1
|   |   Size:                       11404312576 B (11.4 GB)
|   |
|   +-> Volume disk9s1 0BAFEBED-3555-4BF2-B5C5-C5099CA33C46
|       ---------------------------------------------------
|       APFS Volume Disk (Role):   disk9s1 (No specific role)
|       Name:                      WatchOS 11.5 Simulator (Case-insensitive)
|       Mount Point:               /Library/Developer/CoreSimulator/Volumes/watchOS_22T572
|       Capacity Consumed:         11052888064 B (11.1 GB)
|       Sealed:                    Yes
|       FileVault:                 No
```
But Xcode does not:
```
> xcrun simctl runtime list
== Disk Images ==
-- iOS --
iOS 16.4 (20E247) - E3E3AA0C-DBB2-46B8-925D-F405A941B2F3 (Ready)
iOS 26.3.1 (23D8133) - 2399FA1E-C0BC-4B45-8B49-823F5C882563 (Ready)
-- watchOS --
watchOS 26.2 (23S303) - 04EFFC9F-0960-4C2C-96A3-94868360FF03 (Ready)
```

So I'll have to clean manually. Anyway in case you want to ever look for something left over like that.

**Comment by markst** [comment]
I guess the `/System/Library/AssetsV2/com_apple_MobileAsset_iOSSimulatorRuntime` directory can be orphaned too:

<img width="969" height="712" alt="Image" src="https://github.com/user-attachments/assets/2b963423-d6b1-459a-a013-2b3d10d7e3c4" />
